sling/sling.go

```diff
@@ -366,8 +365,7 @@ func (s *Sling) Do(req *http.Request, successV, failureV interface{}) (*http.Res
 
 	// Decode from json
 	if successV != nil || failureV != nil {
-		err = decodeResponseJSON(resp.StatusCode, body, successV, failureV)
-		if err != nil {
+		if err = decodeResponseJSON(resp.StatusCode, body, successV, failureV); err != nil {
```

Comment:
In general I'd rather PRs just contained a single change, rather than mixed in with stylistic changes, but there's nothing too controversial here, so I'll accept it.